internal/plugin/connectors/http/basicauth/plugin.go

```diff
@@ -16,16 +18,22 @@ func PluginInfo() map[string]string {
 	}
 }
 
-// NewConnector returns an http.Connector that decorates each incoming http
-// request with a basic auth header.
-func NewConnector(conRes connector.Resources) http.Connector {
-	return &Connector{
-		logger:   conRes.Logger(),
-	}
-}
-
 // GetHTTPPlugin is required as part of the Secretless plugin spec for HTTP
 // connector plugins. It returns the HTTP plugin.
 func GetHTTPPlugin() http.Plugin {
-	return http.ConnectorConstructor(NewConnector)
+	newConnector, err := generichttp.NewNewConnector(
+		&generichttp.ConfigYAML{
+			CredentialPatterns: map[string]string{
+				"username": "[^:]+",
+			},
+			Headers: map[string]string{
+				"Authorization": " Basic {{ printf \"%s:%s\" .username .password | base64 }}",
+			},
+		},
+	)
+	// This should never occur
+	if err != nil {
+		log.Panicf("Failed to create generic HTTP NewConnector: %s", err)
```

Comment:
Hmm... do we not have our logger passed into these? If not, should we?

Reply:
Ah, notice where it is.  We're inside of `GetHTTPPlugin`, which  returns a _plugin_, which itself provides the `NewConnector` ctor, which is what would be passed the logger.  So this exists "above" anything that would a  logger.  This is framework code.  

To say it another way, you should think of the "generic http" connector as a sort of meta-programming tool (though not in the reflection sense). It allows us to "generate" other plugins "declaratively."  I use scare quotes because it's not code generation as pre-compile step, but conceptually you can think of it code generation.  And it's not declarative in the sense of "a yaml" file, but it is declarative in the sense of "you just define a struct that maps 1-1 to a yaml file."

Reply:
👍 though just a small comment that `It could never occur, unless the connector definition itself was broken` is what the tests should catch and not break the harness. If that code will break the test run, we should make sure that it doesn't.